src/Chainweb/Pact/PactService.hs

```diff
@@ -322,6 +329,50 @@ finalizeCheckpointer finalize = do
 _liftCPErr :: Either String a -> PactServiceM cas a
 _liftCPErr = either internalError' return
 
+-- | Read row from coin-table defined in coin contract, retrieving balance and keyset
+-- associated with account name
+--
+readCoinAccount
+    :: PactDbEnv'
+      -- ^ pact db backend (sqlite)
+    -> Text
+      -- ^ account name
+    -> IO (Maybe (T2 Decimal (P.Guard (P.Term P.Name))))
+readCoinAccount (PactDbEnv' (P.PactDbEnv pdb pdbv)) a = rows >>= \case
+    Nothing -> return Nothfing
+    Just (P.ObjectMap o) -> case Map.toList o of
+      [(P.FieldKey "balance", b), (P.FieldKey "guard", g)] ->
```

Comment:
This is ok for now but it's too strict, i.e. if we decide to add a field to this map we'll have to change this code. Better to do two lookups?

Reply:
if we add a field, we probably want this to fail. It involves changing the coin contract, which would be an entirely new contract with an entirely new table too, since `coin` is non-upgradeable.